### PR TITLE
Update IPAddressParser.cs

### PR DESCRIPTION
**Pull request by Nuget94**
Fix: Remove leading zeros in IPv4 address parsing to prevent octal interpretation

- Added a method `RemoveLeadingZerosFromIPv4` to strip leading zeros from each segment of an IPv4 address before parsing.
- Updated the `Parse` method to use `RemoveLeadingZerosFromIPv4` for IPv4 addresses.
- Ensured that addresses like "023.089.072.068" are correctly interpreted as "23.89.72.68".
- This fix addresses the issue where leading zeros caused segments to be interpreted as octal numbers.

### CODE REVIEW

**Comment by Nuget94** [comment]
@dotnet-policy-service agree

**Comment by huoyaoyuan** [comment]
> * Ensured that addresses like "023.089.072.068" are correctly interpreted as "23.89.72.68".

This is a behavioral change. We need to ensure we want this. For reference, ping.exe on Windows accepts leading zeros in IPv4.

It also need new unit tests for leading zero cases, at src/libraries/System.Net.Primitives/test/.

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddressParser.cs [hunk 1]
```diff
@@ -107,6 +111,36 @@ private static unsafe bool TryParseIPv6(ReadOnlySpan<char> ipSpan, Span<ushort>
             return false;
         }
 
+        private static ReadOnlySpan<char> RemoveLeadingZerosFromIPv4(ReadOnlySpan<char> ipSpan)
+        {
+            Span<char> cleanedIp = stackalloc char[ipSpan.Length];
+            int index = 0;
+            int segmentStart = 0;
```
Comment (huoyaoyuan) [inline src/libraries/System.Net.Primitives/src/System/Net/IPAddressParser.cs:118]: This method is incorrect, and will never work in this form. It returns stack allocated memory out of a method. This is a common mistake in C, and the span lifetime rule of C# blocks this.

Instead, the `TryParseIpv4` method should be updated to accept leading zeros.
Note that the method is currently not optimial.

**Comment by huoyaoyuan** [comment]
> * This fix addresses the issue where leading zeros caused segments to be interpreted as octal numbers.

This would be a breaking change and not something we want to change.

Per https://superuser.com/questions/857603/are-ip-addresses-with-and-without-leading-zeroes-the-same, it's expected that zero-leading IP numbers are explained as octal.

**Comment by Nuget94** [comment]
As [huoyaoyuan](https://github.com/huoyaoyuan) has already written it is probably intended to interpret IP addresses with leading 0 octal therefore no bug that needs a fix.

**Pull request closed without merging**